### PR TITLE
Improvements for Intent.CreateChooser

### PR DESCRIPTION
**Pull request by andrk0p**
### Description
I found that the selection of photos or videos does not work on all devices.

### Actual Behavior

photo_2021-09-03_19-03-29 (2)

### Expected behavior

photo_2021-09-03_19-03-29

The chooserIntent object in all testing cases returns Android.App.Result.Canceled in the onActivityResult method.

I couldn't find a way to get chooserIntent to work correctly on my device, so I added a check for the operating system version to rule out this problem on similar devices.

- Fixed the PickAsync method for devices with Android 11

### API Changes

Added:

```
public bool UseCreateChooser { get; set; } = true;
```

### Configuration
 - Device: Xiaomi Redmi Note 8 Pro
 - OS Version: MIUI Global 12.5.1 Android 11 RP1A
 - Plugin Version: 1.0.2.1

### PR Checklist ###

- [x] All projects build
- [x] Has samples
- [x] Rebased onto current main

### CODE REVIEW

File: MediaGallery/MediaGallery/PickMedia.android.cs [hunk 1]
```diff
@@ -62,7 +61,9 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
                 if (!string.IsNullOrWhiteSpace(request.Title))
                     intent.PutExtra(Intent.ExtraTitle, request.Title);
 
-                chooserIntent = Intent.CreateChooser(intent, request.Title ?? string.Empty);
+                intent = !Platform.HasSdkVersion(30)
```
Comment (dimonovdd) [inline MediaGallery/MediaGallery/PickMedia.android.cs:64]: Thanks. Сan we add a check based on the device model? I tested this plugin on Android 11 devices and it works correctly.

Comment (andrk0p) [reply MediaGallery/MediaGallery/PickMedia.android.cs:64]: Yes, I can add a device model check using ```Xamarin.Essentials.DeviceInfo.Model```

**Comment by dimonovdd** [comment]
@andrk0p I was able to test this on a device like yours but with 12.5.2 version of MIUI. I can't reproduce this problem. Do you have any ideas?

**Comment by andrk0p** [comment]
> 
> 
> @andrk0p I was able to test this on a device like yours but with 12.5.2 version of MIUI. I can't reproduce this problem. Do you have any ideas?

I don't have any ideas. Maybe the problem is in MAUI 12.5.1

**Comment by dimonovdd** [comment]
@andrk0p Maybe, can we add a new parameter for determining use of `Intent.CreateChooser`?

**Comment by andrk0p** [comment]
> 
> 
> @andrk0p Maybe, can we add a new parameter for determining use of `Intent.CreateChooser`?

I added the NeedUseCreateChooser parameter and updated the usage example

File: README.md [hunk 2]
```diff
@@ -130,12 +131,14 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
 - When using the `PickAsync` method the `selectionLimit` parameter just sets multiple pick allowed
 - A request to cancel `PickAsync` method will cancel a task, but the picker UI can remain open until it is closed by the user
 - The use of `Title` property depends on each device
+- `UseCreateChooser` specifies whether to use Intent.createChooser for picking.
```
Comment (dimonovdd) [inline README.md:134]:
```suggestion
- `UseCreateChooser` specifies whether to use `Intent.CreateChooser`
```

File: MediaGallery/MediaGallery/PickMedia.android.cs [hunk 3]
```diff
@@ -62,7 +62,11 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
                 if (!string.IsNullOrWhiteSpace(request.Title))
                     intent.PutExtra(Intent.ExtraTitle, request.Title);
 
-                chooserIntent = Intent.CreateChooser(intent, request.Title ?? string.Empty);
+                if (request.UseCreateChooser)
+                {
+                    intentDisposable = intent;
+                    intent = Intent.CreateChooser(intentDisposable, request.Title ?? string.Empty);
```
Comment (dimonovdd) [inline MediaGallery/MediaGallery/PickMedia.android.cs:68]:
```suggestion
                    intent = Intent.CreateChooser(intent, request.Title ?? string.Empty);
```

File: MediaGallery/MediaGallery/PickMedia.android.cs [hunk 4]
```diff
@@ -26,7 +26,7 @@ static async Task<IEnumerable<IMediaFile>> PlatformPickAsync(MediaPickRequest re
         {
             token.ThrowIfCancellationRequested();
             Intent intent = null;
-            Intent chooserIntent = null;
+            Intent intentDisposable = null;
```
Comment (dimonovdd) [inline MediaGallery/MediaGallery/PickMedia.android.cs:29]:
```suggestion
            IDisposable intentDisposable = null;
```

File: MediaGallery/MediaPickRequest/MediaPickRequest.shared.cs [hunk 5]
```diff
@@ -28,5 +28,8 @@ public MediaPickRequest(int selectionLimit = 1, params MediaFileType[] types)
 
         /// <summary>Gets or sets the source rectangle to display the Share UI from. This is only used on iPad currently.</summary>
         public Rectangle PresentationSourceBounds { get; set; } = default;
+
+        /// <summary>Gets or sets whether to use Intent.CreateChooser for picking. Currently used only for Android.</summary>
```
Comment (dimonovdd) [inline MediaGallery/MediaPickRequest/MediaPickRequest.shared.cs:32]:
```suggestion
        /// <summary>Gets or sets whether to use Intent.CreateChooser. Currently used only for Android.</summary>
```

**Review by dimonovdd — CHANGES_REQUESTED** [review]
LGTM! I added a few minor edits:

**Review by dimonovdd — APPROVED** [review]

**Pull request merged**